src/nvim/strings.c

```diff
@@ -354,7 +354,7 @@ char *strcase_save(const char *const orig, bool upper)
     int l = utf_ptr2len((const char_u *)p);
     if (c == 0) {
       // overlong sequence, use only the first byte
-      c = *p;
+      c = (char_u)(*p);
```

Comment:
seems like it could actually change behaviour, `*p` is likely non-ASCII here. is there any test covering this (strcase_save somehow being called with an overlong)?

Reply:
I took a look, couldn't find any tests on this function.

Reply:
I can revert this change, or alternatively write new tests for this?

Reply:
I don't know what an overlong is. I've tests different alphabets, take a look and see if it's sufficent enough testing or if we need more.

Reply:
It means an UTF-8 sequence which is invalid because there is a compacter representation, like `"\xc0\x80"` would be an overlong representation of NUL, or (perhaps more relevantly) `"\xc1\x81"` would be an overlong `A`